src/node/node_state.h

```diff
@@ -1507,38 +1535,37 @@ namespace ccf
           std::unordered_set<NodeId> configuration =
             consensus->get_latest_configuration();
 
-          for (auto& [node_id, ni] : w)
+          for (const auto& [node_id, opt_ni] : w)
           {
-            if (kv::is_deleted(ni.version))
+            if (!opt_ni.has_value())
             {
-              configuration.erase(node_id);
+              throw std::logic_error("Unexpected: removal from nodes table");
```

Comment:
Not likely this'll happen, but if it does there's a small chance we may be happy to have the node id.